documentation/tutorials/password-change.md

```diff
@@ -0,0 +1,260 @@
+
+Start by defining the module, and defining the template in the `render/1` function.
+```elixir
+# lib/my_app_web/components/change_password_component.ex
+defmodule MyAppWeb.ChangePasswordComponent do
+  @moduledoc """
+  LiveComponent for changing the current user's password.
+  """
+  use MyAppWeb, :live_component
+  alias MyApp.Accounts.User
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div>
+      <.simple_form
+        for={@form}
+        id="user-password-change-form"
+        phx-target={@myself}
+        phx-submit="save"
+      >
+        <.input field={@form[:current_password]} type="password" label="Current Password" />
+        <.input field={@form[:password]} type="password" label="New Password" />
+        <.input field={@form[:password_confirmation]} type="password" label="Confirm New Password" />
+        <:actions>
+          <.button phx-disable-with="Saving...">Save</.button>
+        </:actions>
+      </.simple_form>
+    </div>
+    """
+  end
+end
+```
+
+This will produce a form with the usual three fields (current password, new password, and confirmation of new password, to guard against typographical errors), and a submit button.  Now let's populate the `@form` assign.
+
+```elixir
+  @impl true
+  def update(assigns, socket) do
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign_form()}
+  end
+
+  defp assign_form(%{assigns: %{current_user: user}} = socket) do
+    form =
+      AshPhoenix.Form.for_update(user, :change_password,
+        as: "user",
+        # Only necessary when using field policies with `private_fields :include`.
+        prepare_source: fn changeset ->
+          %{
+            changeset
+            | data:
+                Ash.load!(changeset.data, :hashed_password,
+                  context: %{private: %{password_change?: true}}
+                )
+          }
+        end,
+        actor: user
+      )
+
+    assign(socket, form: to_form(form))
+  end
+```
+`update/2` is covered in the `Phoenix.LiveComponent` life-cycle documentation, so we won't go into it here.  The private function `assign_form/1` should look familiar if you have any forms for Ash resources in your application, but with a significant addition: the `prepare_source` option.
+
+When using field policies, the `@current_user` assign set by AshAuthentication will not contain the value of the `hashed_password` attribute, because it is a private field.  This is what you normally want in your application, but the `:change_password` action needs this to validate the `:current_password` argument, so we need to explicitly load this attribute.  We also pass a private context field to the load operation to use in a field policy that we will write later on.
```

Comment:
```suggestion
When using field policies, the `@current_user` assign set by AshAuthentication may not contain the value of the `hashed_password` attribute, because it is a private field (if you are using `private_fields :hide` or `private_fields :include`).  This is what you normally want in your application, but the `:change_password` action needs this to validate the `:current_password` argument. You may need to explicitly load this attribute.  We also pass a private context field to the load operation to use in a field policy that we will write later on.
```